### PR TITLE
Do not check vmSetName when getting Azure node's IP

### LINKED ISSUE #63538: azure Route Creation is failing when using non Standard LB with VMSS
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
/kind bug
> /kind feature

**What happened**:
Cannot get my Node Networking to complete with using Azure VMSS with Route Table

**What you expected to happen**:
The networking to be configured

**How to reproduce it (as minimally and precisely as possible)**:
Use Azure cloud provider with VmType: vmss and azure route table, use the basic LB.

**Anything else we need to know?**:
Looks like we have a logic that is expecting a non empty vmSetName

*Create route stack*

```
func (az *Cloud) CreateRoute(ctx context.Context, clusterName string, nameHint string, kubeRoute *cloudprovider.Route) error {
    glog.V(2).Infof("create: creating route. clusterName=%q instance=%q cidr=%q", clusterName, kubeRoute.TargetNode, kubeRoute.DestinationCIDR)
    if err := az.createRouteTableIfNotExists(clusterName, kubeRoute); err != nil {
        return err
    }
    targetIP, _, err := az.getIPForMachine(kubeRoute.TargetNode)
```
```
func (az *Cloud) getIPForMachine(nodeName types.NodeName) (string, string, error) {
    return az.vmSet.GetIPByNodeName(string(nodeName), "")
}
```
```
func (ss *scaleSet) GetIPByNodeName(nodeName, vmSetName string) (string, string, error) {
    nic, err := ss.GetPrimaryInterface(nodeName, vmSetName)
    if err != nil {
        glog.Errorf("error: ss.GetIPByNodeName(%s), GetPrimaryInterface(%q, %q), err=%v", nodeName, nodeName, vmSetName, err)
        return "", "", err
    }
```
```
func (ss *scaleSet) GetPrimaryInterface(nodeName, vmSetName string) (network.Interface, error) {
    managedByAS, err := ss.isNodeManagedByAvailabilitySet(nodeName)
    if err != nil {
        glog.Errorf("Failed to check isNodeManagedByAvailabilitySet: %v", err)
        return network.Interface{}, err
    }
    if managedByAS {
        // vm is managed by availability set.
        return ss.availabilitySet.GetPrimaryInterface(nodeName, "")
    }

    ssName, instanceID, vm, err := ss.getVmssVM(nodeName)
    if err != nil {
        glog.Errorf("error: ss.GetPrimaryInterface(%s), ss.getVmssVM(%s), err=%v", nodeName, nodeName, err)
        return network.Interface{}, err
    }

    // Check scale set name.
    // Backends of Standard load balancer could belong to multiple VMSS, so we don't check vmSet for it.
    // (SLY): error is here. since vmSetName is empty
    if !strings.EqualFold(ssName, vmSetName) && !ss.useStandardLoadBalancer() {
        return network.Interface{}, errNotInVMSet
    }
```

as you can see `return az.vmSet.GetIPByNodeName(string(nodeName), "")` pass an empty string for the `vmSetName`
this is the only place where `return az.vmSet.GetIPByNodeName(string(nodeName), "")` is called and it is always with an empty String
the `""` is pass all the way to `GetPrimaryInterface`
which tries to compare it with the extracted `ssName` in `if !strings.EqualFold(ssName, vmSetName)`

I’m using the basic LB

**Environment**:
- Kubernetes version (use `kubectl version`): 1.10.2
- Cloud provider or hardware configuration: azure
- OS (e.g. from /etc/os-release): centos 7.4
- Kernel (e.g. `uname -a`):
- Install tools: in-house (salt-based)
- Others:
/sig azure
/assign @feiskyer

### PR DESCRIPTION
**Pull request by feiskyer**
**What this PR does / why we need it**:

Do not check vmSetName when getting Azure node's IP.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #63538

**Special notes for your reviewer**:

**Release note**:

```release-note
Do not check vmSetName when getting Azure node's IP
```

### CODE REVIEW

File: pkg/cloudprovider/providers/azure/azure_vmss.go [hunk 1]
```diff
@@ -435,9 +435,14 @@ func (ss *scaleSet) GetPrimaryInterface(nodeName, vmSetName string) (network.Int
...
-	// Backends of Standard load balancer could belong to multiple VMSS, so we don't check vmSet for it.
-	if !strings.EqualFold(ssName, vmSetName) && !ss.useStandardLoadBalancer() {
+	// Check scale set name. Note that vmSetName is empty string when getting the
+	// Node's IP address. While vmSetName is not empty, it should be checked with
+	// Node's real scale set name:
+	// - For basic SKU load balancer, errNotInVMSet should be returned if the node's
+	//   scale set is mismatched with vmSetName.
+	// - For standard SKU load balancer, backend could belong to multiple VMSS, so we
+	//   don't check vmSet for it.
+	if vmSetName != "" && !ss.useStandardLoadBalancer() && !strings.EqualFold(ssName, vmSetName) {
```
Comment (karataliu) [inline pkg/cloudprovider/providers/azure/azure_vmss.go]: For scaleSet, is there still any place GetPrimaryInterface is called with vmSetName!=null? Can we just remove the vmSetName parameter. GetPrimaryInterface could be private methods which can be removed from 'VMSet interface'?

Comment (feiskyer) [reply pkg/cloudprovider/providers/azure/azure_vmss.go]: seems not. Let me make it private

Comment (feiskyer) [reply pkg/cloudprovider/providers/azure/azure_vmss.go]: wait, there is a `availalbilitySet` wrap within scaleSet, which requires it within VMSet interface:

https://github.com/kubernetes/kubernetes/blob/5ba149e182da67e386e45910af6d875603a766f0/pkg/cloudprovider/providers/azure/azure_vmss.go#L427-L429

Comment (karataliu) [reply pkg/cloudprovider/providers/azure/azure_vmss.go]: That call also pass "" to vmSetName. Seems the ideal interface here is: GetPrimaryInterface(nodeName string)

The only usage of vmSet in availabilitySet.GetPrimaryInterface is:
https://github.com/kubernetes/kubernetes/blob/b976ebd56a07ebb009aeb8f14913e3472e5609ee/pkg/cloudprovider/providers/azure/azure_standard.go#L578-L585

Can we move this section to 'ensureHostInPool', which is the only place requires vmSet validation. It looks like more the functionality of 'ensureHostInPool'.

**Comment by feiskyer** [comment]
@karataliu Addressed comments. PTAL

**Pull request merged**